zerver/lib/streams.py

```diff
@@ -312,22 +312,22 @@ def access_stream_for_send_message(
 
     # All other cases are an error.
     raise JsonableError(
-        _("Not authorized to send to stream '{channel_name}'").format(channel_name=stream.name)
+        _("Not authorized to send to channel '{channel_name}'").format(channel_name=stream.name)
     )
 
 
 def check_for_exactly_one_stream_arg(stream_id: Optional[int], stream: Optional[str]) -> None:
     if stream_id is None and stream is None:
-        raise JsonableError(_("Please supply 'stream'."))
+        raise JsonableError(_("Please supply a channel parameter: name or ID."))
 
     if stream_id is not None and stream is not None:
-        raise JsonableError(_("Please choose one: 'stream' or 'stream_id'."))
+        raise JsonableError(_("Please supply only one channel parameter: name or ID."))
```

Comment:
The only endpoint currently using these errors is the deprecated mute topic endpoint (https://zulip.com/api/mute-topic), so I'm not super excited about creating a new string to translate for that.

The first one could maybe be simplified to "Missing parameter"? I'm not sure about the second one though.

Reply:
Yeah agreed; the first one could just use the standard missing parameter exception for the `stream_id` field, since the name use case is going to be less common.

And then maybe we should figure out a common translateable exception in `zerver/lib/exceptions` for "Unsupported parameter combination: {parameters}". The Pydantic migration may offer better tooling for this, but we're generally on the hook for own strings, so maybe a quick "api design" thread for this would be prudent, and maybe just clean it up in a little side PR.

Reply:
Started a conversation here: https://chat.zulip.org/#narrow/stream/378-api-design/topic/error.20for.20unsupported.20parameter.20combination/near/1785384

Reply:
For both parameters missing, I updated this to use the `REQ` error string for a missing required parameter: `Missing '{var_name}' argument".

Reply:
Great!